dissect/target/plugins/filesystem/ntfs/utils.py

```diff
@@ -20,13 +23,33 @@ def get_drive_letter(target: Target, filesystem: NtfsFilesystem):
 
     When the drive letter is not available for that filesystem it returns empty.
     """
+    # A filesystem can be known under multiple drives (mount points). If it is
+    # a windows system volume, there are the default sysvol and c: drives.
+    # If the target has a virtual ntfs filesystem, e.g. as constructed by the
+    # tar and dir loaders, there is also the /$fs$/fs<n> drive, under which the
+    # "fake" ntfs filesystem is mounted.
+    # The precedence for drives is first the drive letter drives (e.g. c:),
+    # second the "normally" named drives (e.g. sysvol) and finally the anonymous
+    # drives (e.g. /$fs/fs0).
     mount_items = (item for item in target.fs.mounts.items() if hasattr(item[1], "ntfs"))
```

Comment:
```suggestion
    mount_items = (item for item in target.fs.mounts.items() if item[1].__type__ ==  "ntfs")
```

Reply:
This will break dir and tar filesystems with a fake ntfs part though.

If the mft plugin walks the actual NtfsFilesystem (which checks on `__type__ == "ntfs"`). The paths generated from this will get an anonymous fs mountpoint (e.g. `$fs$\fs0`), but these paths are not reachable through things like target-shell, as the directory structures etc are not available, just the mft.

Reply:
Pff right, forgot about that and thought we'd already fixed that.